controllers/components/dashboard/dashboard_controller.go

```diff
@@ -108,6 +108,8 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 			deploy.WithCache(),
 		)).
 		WithAction(deployments.NewAction()).
+		// TODO: to be removed: https://issues.redhat.com/browse/RHOAIENG-21080
+		WithAction(patchOdhDashboardConfig).
```

Comment:
My understanding was that this were supposed to be a one off script, so a task for https://github.com/opendatahub-io/opendatahub-operator/blob/main/pkg/upgrade/upgrade.go. 

Done as an action would mean that this process would kick in for every reconciliation.
Not sure if this the intended behavior, @csams WDYT ?

Reply:
Absolutely only do this once on upgrade -- never again after that -- we are relying on it to be removable (you folks set the state, we then allow the customer to migrate away from a finite set of data)

@Gkrumbach07 please make sure to work with @AjayJagan so this aligns with our expectations.

Reply:
cool! Thanks @andrewballantyne. @lburgazzoli I will make the changes accordingly